cmd/replication/main.go

```diff
@@ -57,16 +63,19 @@ func main() {
 	var wg sync.WaitGroup
 	doneC := make(chan bool, 1)
 	tracing.GoPanicWrap(ctx, &wg, "main", func(ctx context.Context) {
-		db, err := db.NewNamespacedDB(
-			ctx,
-			options.DB.WriterConnectionString,
-			utils.BuildNamespace(options),
-			options.DB.WaitForDB,
-			options.DB.ReadTimeout,
-		)
-
-		if err != nil {
-			logger.Fatal("initializing database", zap.Error(err))
+		var dbInstance *sql.DB
+		if options.Replication.Enable || options.Sync.Enable || options.Indexer.Enable {
```

Comment:
I'm wondering if it makes sense to use a singleton pattern for the DB, and just instantiate it wherever it is needed, rather than trying to comprehensively enumerate all the cases here.

Or: Just instantiate the DB if the DB params are specified, and don't instantiate if it's not, and throw an error if a DB is used somewhere when it hasn't been instantiated.

Reply:
I will give it some thought tomorrow!